gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimestampBasedCopyableDataset.java

```diff
@@ -134,7 +133,11 @@ public TimestampBasedCopyableDataset(FileSystem fs, Properties props, Path datas
     } finally {
       ExecutorsUtils.shutdownExecutorService(executor, Optional.of(log));
     }
-    return copyableFileList;
+
+    ConcurrentLinkedQueue<CopyableFile> copyableFilesFilteredList = new ConcurrentLinkedQueue<>();
```

Comment:
Do we need ConcurrentLinkedQueue? Seems like List<CopyableFile> should suffice?

Reply:
Also, see the comment earlier about returning a merged path filter in TimeStampBasedCopyableDataset#copyFileFilter() method. That way, you can remove this filtering logic in the end.

Reply:
Existing contract returns ConcurrentLinkedQueue object, therefore did not change the object type.